pywbem/tupleparse.py

```diff
@@ -103,15 +105,20 @@ def filter_tuples(list_):
 
 
 def pcdata(tup_tree):
-    """Return the concatenated character data within a tup_tree.
+    """
+    Return the concatenated character data within the child nodes of a
+    tuple tree node, as a unicode string. Whitespace is preserved.
 
-    The tup_tree must not have non-character children."""
+    The child nodes must be text nodes (no element nodes).
+    """
     for inst in tup_tree[2]:
         if not isinstance(inst, six.string_types):
             raise ParseError("Element %r has unexpected child elements: %r"
                              "(allowed is only text content)" %
                              (name(tup_tree), inst))
-    return ''.join(tup_tree[2])
+    data = u''.join(tup_tree[2])
+    assert isinstance(data, six.text_type)
```

Comment:
Question? Leave this assert in permanently???

Reply:
My vote is yes. On my end, there is a small remaining uncertainty about the guarantee that the sax parser always returns unicode, and if pywbem wants to guarantee that from that point on, we need to either ensure that ourselves by check&conversion (e.g. using `_ensure_unicode()`) or by verifying it with an assertion.

About the uncertainty: The docs about [ContentHandler.characters()](https://docs.python.org/2/library/xml.sax.handler.html#xml.sax.handler.ContentHandler.characters) state: "content may be a Unicode string or a byte string; the expat reader module produces always Unicode strings." but I did not find a statement that the expat reader is used. This is implemented in C, so I did not venture to have a look.

Reply:
AGREED